### PR TITLE
Added forts

### PR DESCRIPTION
**Pull request by BenBooth1344**

### CODE REVIEW

File: core/src/com/unciv/UnCivGame.kt [hunk 1]
```diff
@@ -25,10 +25,10 @@ class UnCivGame(val version: String) : Game() {
      * This exists so that when debugging we can see the entire map.
      * Remember to turn this to false before commit and upload!
      */
-    var viewEntireMapForDebug = false
+    var viewEntireMapForDebug = true
```
Comment (yairm210) [inline core/src/com/unciv/UnCivGame.kt]: Not for commiting please

File: core/src/com/unciv/logic/battle/BattleDamage.kt [hunk 2]
```diff
@@ -15,6 +15,11 @@ class BattleDamageModifier(val vs:String,val modificationAmount:Float){
     fun getText(): String = "vs $vs"
 }
 
+enum class CombatantRole{
```
Comment (yairm210) [inline core/src/com/unciv/logic/battle/BattleDamage.kt]: TBH I don't really like this, using n enum where a boolean true/false would suffice, but OK I guess

**Comment by yairm210** [comment]
If the AI doesn't use forts then it's basically a human cheat...

**Comment by BenBooth1344** [comment]
As referenced in the following link, the AI apparently basically doesn't use forts in Civ V
https://forums.civfanatics.com/threads/does-the-ai-build-forts.444769/

**Comment by BenBooth1344** [comment]
I'm not sure the best way to get the AI to use the forts. The best I can think of is to have AI workers start following around combat units building forts after the worker can no longer find anything else productive to do..

**Comment by yairm210** [comment]
2 points:
1. Fort should have a descriptive 'unique' in json, see moai, for 2 reasons: it lets us have unique improvements that replace the fort, and will allow users to know what the fort does
2. Fort 'defensive bonus' should be where the tile defensive bonus is calculated, it replaces it

**Comment by yairm210** [comment]
Unique text: "Gives a defensive bonus of 50%"

**Comment by yairm210** [comment]
?

**Comment by yairm210** [comment]
#2325 adds this functionality, with AI - thanks for implementing the basics! =D

**Pull request closed without merging**